pycqed/analysis_v2/base_analysis.py

```diff
@@ -234,13 +242,50 @@ def run_analysis(self):
                 if self.options_dict.get('save_figs', False):
                     self.save_figures(close_figs=self.options_dict.get(
                         'close_figs', False))
+            self._raise_warning()
         except Exception as e:
             if self.raise_exceptions:
                 raise e
             else:
                 log.error("Unhandled error during analysis!")
                 log.error(traceback.format_exc())
 
+    def _raise_warning(self):
+        """
+        If delegate_plotting is False, calls raise_warning_image which saves a
+        warning image to the folder corresponding to the last timestamp in
+        self.timestamps.
+
+        If warning_message is not None, calls write_warning_message_to_text_file
+        which creates a text file with warning_message in the folder
+        corresponding to the last timestamp in self.timestamps. If text file
+        already exists, it will append to it.
+
+        Params that can be passed in the options_dict:
+        :param warning_message: string with the message to be written into the
+            text file. self.warning_message will be appended to this
+        :param warning_textfile_name: string with name of the file without
+            extension.
+        """
+        if not self.check_plotting_delegation():
+            # else, the warning image and text file will be generated twice
+            # when the AnalysisDaemon is active
```

Comment:
Just some general comments (do not necessarily need to be addressed in this PR):
- Explaining an if statement with a comment that starts with the word "else" might cause confusion.
- I wonder what the best place for such an explanation is: in the docstring or here.
- I would prefer the following code in such an instance to avoid an additional level of indentation. It does not really matter in this special case since there are no deep indentation levels in this function, so it is, as I said, just a general comment
```
if self.check_plotting_delegation():
    return
```